ansible/roles/compute_init/tasks/main.yml

```diff
@@ -0,0 +1,133 @@
+---
+
+- name: Ensure directories exist
```

Comment:
So we also need to inject our group vars. Except actually group vars don't really exist once in a play. So we could create hostvars like `.../inventory/$host.yml`, but then file actually has to be [host-specific](https://docs.ansible.com/ansible/latest/inventory_guide/intro_inventory.html#assigning-a-variable-to-one-machine-host-variables):

```
all:
  {{ inventory_hostname }}:
     {{ hostvars[hostname] | sort_out_indentation }}
```
and we have to somehow make "localhost" *that* host.

Maybe there's a better way ....

Reply:
Actually its not that bad. The only host in a compute_init play is localhost, so we can create group_vars/all for each host. E.g we template out `..$host/inventory/group_vars/all.yml` but its is literally just `{{ hostvars[hostname] }}`.

The other alternative is just that that templated out file is read using `include_vars:`.

I'm not sure which we should choose; what we *do* need is that we preserve the "normal ansible" behaviour where group_vars override role defaults; if the compute_init "role default" equivalents are in `vars:` blocks the nfs-exported vars must override those. See the ansible variable precedence docs ...